Allow $PANTS_HOME to be a relative path

Eric-Arellano opened the pull request:
A user requested this because their CI provider GitLab works better with relative paths than absolute paths. Generally, Pants allows relative paths, so it's consistent for us to support them here, too.

This also makes some other improvements:
* Fix our version check to look for >= 1.23, rather than > 1.16. This has been the case, but we didn't change the version check.
* Use the more inclusive term `smoke screen`.

Eric-Arellano committed:
Allow PANTS_HOME to be a relative path

Eric-Arellano committed:
Grammar

stuhood commented on pants:
```diff
@@ -116,8 +121,8 @@ function determine_pants_version {
+  if [[ "${pants_major_version}" -eq 1 && "${pants_minor_version}" -lt 23 ]]; then
+    die "This version of the \`./pants\` script does not work with Pants < 1.23.0. Instead,
 either upgrade your \`pants_version\` or use the version of the \`./pants\` script at
 https://raw.githubusercontent.com/pantsbuild/setup/3bb006e4a792ae83d7059ea76c0372ece7c1069d/pants."
```
Should this now recommend a "more recent old version"?

Eric-Arellano replied:
I'm not sure how to handle it. The most robust approach would be to have two checks. First check if < 16, and stick with this message. Then, check if >=16 and < 23, then do a new message.

Thoughts?

stuhood replied:
It would probably be ok to walk a linked list of "try this script, it will recommend an older one, rinse repeat".

stuhood reviewed (approved):
Thanks!

Eric-Arellano committed:
If >1.16 but <1.23, use a more modern version of the script

Also fix a test

Eric-Arellano committed:
Only check for <= 1.22, not <= 1.16

Stu pointed out that the script for <= 1.22 will complain if they're still using too old of a script.
